### ISSUE TITLE
[Bug][a11y]: deletable tags must support deleting a tag using delete or backspace key

### ISSUE THREAD

**Issue by dineshy87** [adobe/spectrum-web-components#3364]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-tags

### Expected behavior

Spectrum design specifies that users should be able to press delete or backspace button to delete a deletable tag. Once implemented, make sure the keyboard focus moves on to the next non-disabled tag within the sp-tags.

For reference, check [react-spectrum implementation](https://react-spectrum.adobe.com/react-spectrum/TagGroup.html#contextual-help)

### Actual behavior

The above behavior is currently not supported. 

### What browsers are you seeing the problem in?

All

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/components/tags/
2. Try deleting tags using keyboard.

Related issue:
https://jira.corp.adobe.com/browse/CSS-552

**Comment by Rajdeepc** [comment]
PR: https://github.com/adobe/spectrum-web-components/pull/3391

**Comment by blunteshwar** [comment]
The `mutationController` is calling the callback fucntion `handleItemMutaion` whenever there is a mutation in the host element whether the mutation is of the type `delete`, `add` or `update`. Thus some tests of sp-grid are failing
